controllers/mlmd_test.go

```diff
@@ -0,0 +1,249 @@
+
+	// Create Context, Fake Controller and Params
+	ctx, params, reconciler := CreateNewTestObjects()
+	params.ExtractParams(ctx, dspa, reconciler.Client, reconciler.Log)
+
+	// Ensure MLMD-Envoy resources doesn't exist
+	deployment := &appsv1.Deployment{}
+	created, err := reconciler.IsResourceCreated(ctx, deployment, expectedMLMDEnvoyName, testNamespace)
+	assert.False(t, created)
+	assert.Nil(t, err)
+
+	// Ensure MLMD-GRPC resources doesn't exist
+	deployment = &appsv1.Deployment{}
+	created, err = reconciler.IsResourceCreated(ctx, deployment, expectedMLMDGRPCName, testNamespace)
+	assert.False(t, created)
+	assert.Nil(t, err)
+
+	// Ensure MLMD-Writer resources doesn't exist
+	deployment = &appsv1.Deployment{}
+	created, err = reconciler.IsResourceCreated(ctx, deployment, expectedMLMDWriterName, testNamespace)
+	assert.False(t, created)
+	assert.Nil(t, err)
+
+	// Run test reconciliation
+	err = reconciler.ReconcileMLMD(dspa, params)
+	assert.Nil(t, err)
+
+	// Ensure MLMD-Envoy resources doesn't exist
+	deployment = &appsv1.Deployment{}
+	created, err = reconciler.IsResourceCreated(ctx, deployment, expectedMLMDEnvoyName, testNamespace)
+	assert.True(t, created)
+	assert.Nil(t, err)
+
+	// Ensure MLMD-GRPC resources doesn't exist
+	deployment = &appsv1.Deployment{}
+	created, err = reconciler.IsResourceCreated(ctx, deployment, expectedMLMDGRPCName, testNamespace)
+	assert.True(t, created)
+	assert.Nil(t, err)
+
+	// Ensure MLMD-Writer resources doesn't exist
+	deployment = &appsv1.Deployment{}
+	created, err = reconciler.IsResourceCreated(ctx, deployment, expectedMLMDWriterName, testNamespace)
+	assert.True(t, created)
+	assert.Nil(t, err)
+}
+
+func TestDontDeployMLMD(t *testing.T) {
+	testNamespace := "testnamespace"
+	testDSPAName := "testdspa"
+	expectedMLMDEnvoyName := "ds-pipeline-metadata-envoy-testdspa"
+	expectedMLMDGRPCName := "ds-pipeline-metadata-grpc-testdspa"
+	expectedMLMDWriterName := "ds-pipeline-metadata-writer-testdspa"
+
+	// Construct MLMD Spec
+	dspa := &dspav1alpha1.DataSciencePipelinesApplication{
+		Spec: dspav1alpha1.DSPASpec{
+			APIServer: &dspav1alpha1.APIServer{
+				ArchiveLogs: true, // TODO: This appears to be required which is out-of-spec (.Spec.APIServer should be fully defaultable), but test throws an nil pointer panic if it isn't provided. possibly due to test setup - Investigate.
+			},
+			MLMD: &dspav1alpha1.MLMD{
+				Deploy: false,
+			},
+			Database: &dspav1alpha1.Database{
+				DisableHealthCheck: false,
+				MariaDB: &dspav1alpha1.MariaDB{
+					Deploy: true,
+				},
+			},
+			ObjectStorage: &dspav1alpha1.ObjectStorage{
+				DisableHealthCheck: false,
+				Minio: &dspav1alpha1.Minio{
+					Deploy: false,
+				},
+			},
+		},
+	}
+
+	// Enrich DSPA with name+namespace
+	dspa.Namespace = testNamespace
+	dspa.Name = testDSPAName
+
+	// Create Context, Fake Controller and Params
+	ctx, params, reconciler := CreateNewTestObjects()
+	params.ExtractParams(ctx, dspa, reconciler.Client, reconciler.Log)
```

Comment:
`ExtractParams` returns an error, we should assert on this (same for everywhere this is used)